File: src/main/java/com/moovie/backend/domain/usecase/GetCommentsUseCase.java
package com.moovie.backend.domain.usecase;

import com.moovie.backend.domain.repository.CommentsRepository;
import com.moovie.backend.domain.model.CommentListing;
import org.springframework.stereotype.Service;

@Service
public class GetCommentsUseCase {

    private final CommentsRepository commentsRepository;

    public GetCommentsUseCase(CommentsRepository commentsRepository) {
        this.commentsRepository = commentsRepository;
    }

    public CommentListing invoke(String contentId, int page, int pageSize) {
        if (contentId == null || contentId.isBlank()) {
            throw new IllegalArgumentException("contentId must not be blank");
        }
        if (page < 1) {
            throw new IllegalArgumentException("page must be >= 1");
        }
        if (pageSize < 1) {
            throw new IllegalArgumentException("pageSize must be >= 1");
        }
        return commentsRepository.getComments(contentId, page, pageSize);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/GetCommentsUseCase.kt
package org.mobyle.domain.usecase

import org.mobyle.domain.model.CommentListing
import org.mobyle.domain.repository.CommentsRepository

class GetCommentsUseCase(
    private val commentsRepository: CommentsRepository
) {
    suspend operator fun invoke(contentId: String, page: Int = 1, pageSize: Int = 10): CommentListing {
        require(contentId.isNotBlank()) { "Content ID cannot be blank" }
        require(page > 0) { "Page must be greater than 0" }
        require(pageSize > 0) { "Page size must be greater than 0" }

        return commentsRepository.getComments(contentId, page, pageSize)
    }
}
